utils/hct/hctbuild.cmd

```diff
@@ -465,10 +465,8 @@ if "%DO_SETUP%"=="1" (
     echo Running "%CMAKE_PATH%" -DCMAKE_BUILD_TYPE:STRING=%1 %CMAKE_OPTS% -G %4 %HLSL_SRC_DIR% > %3\cmake-log.txt
     "%CMAKE_PATH%" -DCMAKE_BUILD_TYPE:STRING=%1 %CMAKE_OPTS% -G %4 %HLSL_SRC_DIR% >> %3\cmake-log.txt 2>&1
   ) else (
-    rem -DCMAKE_BUILD_TYPE:STRING=%1 is not necessary for multi-config generators like VS
-    rem but need CMAKE_BUILD_TYPE to generate lit cfg.
     echo Running "%CMAKE_PATH%" -DCMAKE_BUILD_TYPE:STRING=%1  %CMAKE_OPTS% -G %4 %5 %HLSL_SRC_DIR% > %3\cmake-log.txt
-    "%CMAKE_PATH%" %CMAKE_OPTS% -G %4 %5 %HLSL_SRC_DIR% >> %3\cmake-log.txt 2>&1
```

Comment:
Maybe add comment about why we need CMAKE_BUILD_TYPE for multi-config generators like VS?

Reply:
Might be a long comment! I tried to be succinct here and failed. I feel like the explanation for why it wasn't needed was more comment worthy than the fact that we do. Maybe I could just say that it will still generate multiple versions in spite of that.

Reply:
"BUILD_TYPE is mostly ignored in this path as VS generates multiple targets. It is still needed to satisfy cmake file expectations"

I feel pretty good about that. What do you think?

Reply:
FWIW, the latest version of LLVM CMakeLists fails if you don't set this flag, which avoids the debug by default that was the first step in this walking catastrophe

Reply:
Looks good.
I guess upstream hit the same issue :)